lib/ng2-select2.component.ts

```diff
@@ -95,13 +112,37 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
         }
 
         this.element.on('select2:select select2:unselect', () => {
+            this.onChange(that.element.val());
+            this.onTouched();
             this.valueChanged.emit({
-                value: this.element.val(),
-                data: this.element.select2('data')
```

Comment:
That's because it is called when the select2 control has not been constructed yet.

I think the ControlValueAccessor implementation is not working as it should, this needs to be re-thought. Right now it doesn't improve the situation I fear.